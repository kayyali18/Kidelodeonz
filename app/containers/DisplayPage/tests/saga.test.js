/**
 * Tests for DisplayPage sagas
 */

import { put, takeLatest } from 'redux-saga/effects'

import { RUN_STUMBLE_SAGA } from '../constants'
import stumblingSaga, { getStumble } from '../saga'
import { displayStumble } from '../actions'

const stumble = false

/* eslint-disable redux-saga/yield-effects prettier-ignore */
describe('getStumble Saga', () => {
  let getStumbleGenerator

  beforeAll(() => {
    getStumbleGenerator = getStumble()

    // describe the selector (makeSelectData)
    const selectDescriptor = getStumbleGenerator.next()
    expect(selectDescriptor).toMatchSnapshot()
  })

  it('should dispatch the displayStumble with a stumble', () => {
    const putDescriptor = getStumbleGenerator.next(stumble).value
    expect(putDescriptor).toEqual(put(displayStumble(stumble)))
  })
})

describe('stumblingSaga', () => {
  const stumblingSagaTest = stumblingSaga()

  it('should start a task to watch for RUN_STUMBLE_SAGA', () => {
    const takeLatestDescriptor = stumblingSagaTest.next().value
    expect(takeLatestDescriptor).toEqual(
      takeLatest(RUN_STUMBLE_SAGA, getStumble),
    )
  })
})
