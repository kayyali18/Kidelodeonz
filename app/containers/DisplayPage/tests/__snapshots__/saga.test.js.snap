// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;
